### PR TITLE
Fix handling of ARF files

### PR DESCRIPTION
**Pull request by matejak**
In the past, ARFs were used exclusively as a source of Bash/Ansible remediations.
However, it is more practical to have them available as test artifacts, as they contain more information. In order to finish what #7750 just started, following needed to be done:

- Rename `..._file` to `..._basename`, so it is clear what the variable represents.
- Always generate ARFs during initial stage of tests even if reports are not requested - those files are still needed to act as sources of remediations.
- Keep those ARFs that could be used in remediations around for the next stage of tests.
- Make test runner a context manager, which allows it to clean remaining files after the test run finishes.

### CODE REVIEW

**Comment by mildas** [comment]
`--remediate-using ansible` doesn't work
```
$ python3 tests/test_suite.py rule --libvirt qemu:///session test-suite-rhel9 --datastream build/ssg-rhel9-ds.xml --scenarios service_disabled --remediate-using ansible service_pcscd_enabled
Setting console output to log level INFO        
INFO - The base image option has not been specified, choosing libvirt-based test environment.
INFO - Logging into /home/mlysonek/SCAP/content/logs/rule-custom-2022-01-04-1750/test_suite.log
INFO - xccdf_org.ssgproject.content_rule_service_pcscd_enabled            
INFO - Script service_disabled.fail.sh using profile (all) OK   
WARNING - Nothing has been tested!                                                                                     
Traceback (most recent call last):
  File "/home/mlysonek/SCAP/content/tests/test_suite.py", line 462, in <module>
    main()
  File "/home/mlysonek/SCAP/content/tests/test_suite.py", line 458, in main
    options.func(options)
  File "/home/mlysonek/SCAP/content/tests/ssg_test_suite/rule.py", line 498, in perform_rule_check
    checker.test_target(options.target)
  File "/home/mlysonek/SCAP/content/tests/ssg_test_suite/oscap.py", line 681, in test_target
    self._test_target(target)
  File "/home/mlysonek/SCAP/content/tests/ssg_test_suite/rule.py", line 327, in _test_target
    self.test_rule(state, rule, sliced_scenarios_by_rule_id[rule.id])
  File "/home/mlysonek/SCAP/content/tests/ssg_test_suite/rule.py", line 282, in test_rule
    self._check_rule(
  File "/home/mlysonek/SCAP/content/tests/ssg_test_suite/rule.py", line 399, in _check_rule
    state.map_on_top(self._check_and_record_rule_scenario, args_list)
  File "/home/mlysonek/SCAP/content/tests/ssg_test_suite/test_env.py", line 26, in map_on_top
    function(* args_list[0])
  File "/home/mlysonek/SCAP/content/tests/ssg_test_suite/rule.py", line 411, in _check_and_record_rule_scenario
    self._check_rule_scenario(scenario, remote_rule_dir, rule_id, remediation_available)
  File "/home/mlysonek/SCAP/content/tests/ssg_test_suite/rule.py", line 467, in _check_rule_scenario
    self.run_test_for_all_profiles(profiles, test_data)
  File "/home/mlysonek/SCAP/content/tests/ssg_test_suite/oscap.py", line 697, in run_test_for_all_profiles
    self._run_test(profiles[0], test_data)
  File "/home/mlysonek/SCAP/content/tests/ssg_test_suite/rule.py", line 163, in _run_test
    if not self._remediation_went_ok(runner, rule_id):
  File "/home/mlysonek/SCAP/content/tests/ssg_test_suite/rule.py", line 199, in _remediation_went_ok
    success = runner.run_stage_with_context('remediation', 'fixed')
  File "/home/mlysonek/SCAP/content/tests/ssg_test_suite/oscap.py", line 587, in run_stage_with_context
    return self.run_stage(stage)
  File "/home/mlysonek/SCAP/content/tests/ssg_test_suite/oscap.py", line 368, in run_stage
    result = self.remediation()
  File "/home/mlysonek/SCAP/content/tests/ssg_test_suite/oscap.py", line 661, in remediation
    success = run_stage_remediation_ansible('rule', self.environment, formatting, self.verbose_path)
  File "/home/mlysonek/SCAP/content/tests/ssg_test_suite/oscap.py", line 166, in run_stage_remediation_ansible
    returncode, output = common.run_cmd_local(command, verbose_path)
  File "/home/mlysonek/SCAP/content/tests/ssg_test_suite/common.py", line 167, in run_cmd_local
    returncode, output = _run_cmd(command, verbose_path, env)
  File "/home/mlysonek/SCAP/content/tests/ssg_test_suite/common.py", line 176, in _run_cmd
    output = subprocess.check_output(
  File "/usr/lib64/python3.10/subprocess.py", line 420, in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
  File "/usr/lib64/python3.10/subprocess.py", line 501, in run
    with Popen(*popenargs, **kwargs) as process:
  File "/usr/lib64/python3.10/subprocess.py", line 966, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/usr/lib64/python3.10/subprocess.py", line 1842, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'ansible-playbook'
```

`--remediate-using bash` is fine

**Comment by marcusburghardt** [comment]
I also tested this PR locally, where I was able to reproduce the initial problem and it was fixed after this patch.

**Review by marcusburghardt — COMMENTED** [review]
LGTM. Problem was fixed in my test environment after this patch. : )

File: tests/ssg_test_suite/oscap.py [hunk 1]
```diff
@@ -617,6 +634,10 @@ def final(self):
 
 
 class BashRuleRunner(RuleRunner):
+    def initial(self):
+        self.command_options += ['--results-arf', self.arf_path]
+        return super(RuleRunner, self).initial()
```
Comment (mildas) [inline tests/ssg_test_suite/oscap.py:639]:
```suggestion
        return super(BashRuleRunner, self).initial()
```

**Review by mildas — APPROVED** [review]
LGTM, thank you!

**Pull request merged**